doc/rfcs/003-quickly-collaborate-with-coworkers-and-friends.md

```diff
@@ -0,0 +1,146 @@
+3. Host switches from Atom to third-party communication tool (e.g., Slack, IRC)
+4. Host pastes portal ID into third-party communication tool
+5. Guest selects portal ID and copies it to their clipboard
+6. Guest switches from third-party communication tool to Atom
+7. Guest clicks "Join a portal" link or invokes "Join portal" command, and Atom automatically fetches the portal ID from the clipboard
+8. Guest clicks "Join"
+
+With the ability to join a portal via a URL, we can reduce the guest's set-up process from 4 steps to just 1 step:
+
+1. Host shares a portal
+2. Host copies portal URL to their clipboard
+3. Host switches from Atom to third-party communication tool (e.g., Slack, IRC)
+4. Host pastes portal URL into third-party communication tool
+5. Guest clicks portal URL, and the operating system hands control to Atom, and Atom joins the portal
+
+With the ability to invite past collaborators to your portal from within Atom, we reduce the host's set-up process from 4 steps to 3 steps:
+
+1. Host shares a portal
+2. Host selects person in their list of past collaborators
+3. Host invites person to join the portal
+4. Guest accepts invitation to join portal, and Atom joins the portal
+
+And perhaps we can reduce the host's set-up process even further:
+
+1. Host selects a person in their list of past collaborators
+2. Host invites the selected person to a portal, and Atom automatically shares a portal for the host and invites the selected person
+3. Guest accepts invitation to join portal, and Atom joins the portal
+
+## Explanation
+
+### Share a portal via URL
+
+Once a host creates a portal (e.g., by clicking the "share" toggle in Teletype's status bar popover), Teletype presents the host with the portal's URL. The host copies that URL to their clipboard (in the same way that they previously copied the portal ID).
+
+#### Early mockup
+
+<img width="314" src="https://user-images.githubusercontent.com/2988/37489877-6d0b0cfe-286f-11e8-87a3-a04da42994a3.png">
+
+Also consider [Slack's approach](https://github.com/atom/teletype/issues/109#issuecomment-337223084) for potential inspiration.
+
+#### Portal URL format
+
+Prior Teletype releases established a URI structure for identifying editors in Atom's workspace: `atom://teletype/portal/<portal-id>/editor/<editor-id>`
+
+Portal URLs use that same pattern, but exclude the editor ID: `atom://teletype/portal/<portal-id>`
+
+The `<portal-id>` is the same ID used in prior releases for sharing and joining a portal.
+
+Example: `atom://teletype/portal/63b120f3-b646-4c46-8962-656518249186`
+
+### Join a portal via URL
+
+When a guest follows the URL (e.g., by clicking on the URL in Slack, IRC, etc.), Atom opens, and Teletype asks the user if they want to join the portal. If the user chooses to join the portal, they become a guest in the portal (just as they previously did by entering the portal ID and clicking "Join"). If they choose not to join the portal, nothing happens.
```

Comment:
Should we keep the "Join a portal" button (and input when clicked) in the popup? With clickable URLs it's not really needed, except maybe if the URL handling doesn't work for some reason. Like a mis-configured URL handler. Then there is always a fallback of manually pasting the URL. Is that a big enough reason?

Reply:
@simurai: I've been thinking about that, too. Some users may choose _not_ to have Atom handle `atom://...` URLs. For that reason, I do think we should keep the "Join a portal" button and input in the popup. And I think you should be able to paste the URL into that input field in order to join the portal. However, when you click the "Join a portal" button in the popup, we may want to include a message explaining the easier option (i.e., let Atom handle the URL for you).

Reply:
Based on Community & Safety's belief in [consensual software](https://medium.com/consensual-software/consensual-software-how-to-prioritize-user-safety-369f5a4dee8), I would strongly suggest keeping the "Join a portal" button to ensure the user's explicit consent, particularly since there are some privacy concerns mentioned below